src/coreclr/pal/src/exception/signal.cpp

```diff
@@ -167,17 +170,22 @@ BOOL SEHInitializeSignals(CorUnix::CPalThread *pthrCurrent, DWORD flags)
            see sigaction man page for more details
            */
         handle_signal(SIGILL, sigill_handler, &g_previous_sigill);
-        handle_signal(SIGTRAP, sigtrap_handler, &g_previous_sigtrap);
         handle_signal(SIGFPE, sigfpe_handler, &g_previous_sigfpe);
         handle_signal(SIGBUS, sigbus_handler, &g_previous_sigbus);
-        // SIGSEGV handler runs on a separate stack so that we can handle stack overflow
-        handle_signal(SIGSEGV, sigsegv_handler, &g_previous_sigsegv, SA_ONSTACK);
+        handle_signal(SIGABRT, sigabrt_handler, &g_previous_sigabrt);
         // We don't setup a handler for SIGINT/SIGQUIT when those signals are ignored.
         // Otherwise our child processes would reset to the default on exec causing them
         // to terminate on these signals.
         handle_signal(SIGINT, sigint_handler, &g_previous_sigint, 0 /* additionalFlags */, true /* skipIgnored */);
         handle_signal(SIGQUIT, sigquit_handler, &g_previous_sigquit, 0 /* additionalFlags */, true /* skipIgnored */);
 
+#if HAVE_MACH_EXCEPTIONS
+        handle_signal(SIGSEGV, sigsegv_handler, &g_previous_sigsegv);
```

Comment:
Does SIGSEGV (and other hardware exception signals) handler get ever called? We handle hardware failures using exception ports on macOS and I don't think it composes with signal handling of the same things.

Reply:
Yes, they do.  The Mach exception ports get called and then the SIGSEGV, etc. get forwarded through the POSIX signal layer to the signal enable that I enabled.  It is also the only way to catch abort() (SIGABORT) and any signal sent from another process like `kill -s SIGILL <pid>` which isn't that important except for SIGTERM.